Cubical/Algebra/Matrix.agda

```diff
@@ -0,0 +1,146 @@
+{-# OPTIONS --cubical --safe #-}
+module Cubical.Algebra.Matrix where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Univalence
+open import Cubical.Functions.FunExtEquiv
+
+import Cubical.Data.Empty as ⊥
+open import Cubical.Data.Nat hiding (_+_)
+open import Cubical.Data.Vec
+open import Cubical.Data.FinData
+open import Cubical.Relation.Nullary
+
+open import Cubical.Structures.CommRing
+
+private
+  variable
+    ℓ : Level
+    A : Type ℓ
+
+-- Equivalence between Vec matrix and Fin function matrix
+
+FinMatrix : (A : Type ℓ) (m n : ℕ) → Type ℓ
+FinMatrix A m n = FinVec (FinVec A n) m
+
+VecMatrix : (A : Type ℓ) (m n : ℕ) → Type ℓ
+VecMatrix A m n = Vec (Vec A n) m
+
+FinMatrix→VecMatrix : {m n : ℕ} → FinMatrix A m n → VecMatrix A m n
+FinMatrix→VecMatrix {m = zero}  _ = []
+FinMatrix→VecMatrix {m = suc m} M = FinVec→Vec (M zero) ∷ FinMatrix→VecMatrix (λ x → M (suc x))
```

Comment:
Have you considered this definition ?
```agda
FinMatrix→VecMatrix M = FinVec→Vec (\ fm -> FinVec→ (\ fn -> M fm fn))
```

Reply:
Nope, but it's a lot nicer than mine and everything typechecks so will push it soon! Good catch